src/ng/directive/ngTranscludeLocals.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name ngTranscludeLocals
+ * @restrict A
+ *
+ * @description
+ * Directive that allows to define local variables for a given transclusion scope.
+ *
+ * It works with {@link ng.directive:ngTransclude ngTransclude} and it
+ * expects an object which define the local varibles for the transcluded template.
+ *
+ * @element ng-transclude
+ *
+ * @param {expression} ngTranscludeLocals an object with pairs of key-value
+ *                                        in which the key is the variable name inside the template
+ *                                        and the value is the value for those variables
+ *
+ * @example
+ * ### Transclude with locals
+ * This example demonstrates basic transclusion of local context into a transcluded template.
+ * <example name="simpleTranscludeExample" module="transcludeLocalsExample">
+ *   <file name="index.html">
+ *     <script>
+ *       angular.module('transcludeLocalsExample', [])
+ *         .component('heroList', {
+ *           transclude: true,
+ *           template: '<div ng-repeat="hero in $ctrl.heros">' +
+ *                       '<ng-transclude ng-transclude-locals="{hero: hero, position:$index+1}"></ng-transclude>' +
+ *                     '</div>',
+ *           controller: function() {
+ *             this.heros = ['Superman','Batman','Spawn'];
+ *           }
+ *         });
+ *     </script>
+ *     <div>
+ *       <ul>
+ *       <hero-list>
+ *         <li>[{{position}}] {{hero}}</li>
+ *       </hero-list>
+ *       </ul>
+ *     </div>
+ *   </file>
+ *   <file name="protractor.js" type="protractor">
+ *     it('should have transcluded all repeated elements with different local', function() {
+ *       var heros = element.all(by.repeater('hero in $ctrl.heros'));
+ *       expect(heros.get(0).getText()).toEqual('[1] Superman');
+ *       expect(heros.get(1).getText()).toEqual('[2] Batman');
+ *       expect(heros.get(2).getText()).toEqual('[3] Spawn');
+ *      });
+ *   </file>
+ * </example>
+
+ */
+var ngTranscludeLocalsMinErr = minErr('ngTranscludeLocals');
+var ngTranscludeLocalsDirective = function() {
+  return {
+    restrict: 'A',
+    link: function ngTranscludeLocalsLink($scope, $element, $attrs) {
+      $scope.$watchCollection($attrs.ngTranscludeLocals, function(newLocals, oldLocals) {
+        var transcludeScope = $scope.$$childHead;
```

Comment:
Right, if not used with a scope that has no childs, it fails.
Adding test and solution (ignore and do nothing).